iotilecore/iotile/core/hw/auth/cli_auth_provider.py

```diff
@@ -0,0 +1,35 @@
+
+import os
+import binascii
+import struct
+import hmac
+import hashlib
+from iotile.core.exceptions import NotFoundError
+from .rootkey_auth_provider import RootKeyAuthProvider
+import getpass
+
+
+class CliAuthProvider(RootKeyAuthProvider):
+    """Key provider implementation that prompt user for password
+
+        root key is derived using pbkdf2
+    """
+    def __init__(self, args=None):
+        if args is None:
+            args = {}
+
+        args['supported_keys'] = [self.UserKey]
+
+        super().__init__(args)
+
+    @classmethod
+    def derive_key(cls, password):
+        return hashlib.pbkdf2_hmac('sha256', password.encode('utf-8'), b'salt', 100000)
```

Comment:
Checking that this isn't a magic number -- why 100000? If there is a reason it's nice to at least comment it so we all remember.